lib/forms.js

```diff
@@ -26,39 +83,42 @@ var forms = {
     if ( self.connections[key] ) {
       return cb(undefined, self.connections[key]);
     }
+    mongoose.Promise = global.Promise;
     var paramValidation = validate(options);
 
     logger.info("Initialising mongo connections");
 
-    function handleConnectionResponse(err, databaseConn) {
+    async.series([
+      function(cb) {
+        logger.debug("valid params", options);
+        paramValidation.has("uri", function(failed) {
+          if (failed) {
+            return cb(new Error("Invalid Params: " + JSON.stringify(failed)));
+          } else {
+            return cb();
+          }
+        });
+      },
+      async.apply(getMongooseConnection, uri, forms.sharedConnections.mongooseConnection),
+      async.apply(getMongoConnection, uri, forms.sharedConnections.databaseConnection)
+    ], function(err, results) {
```

Comment:
`results` is very generic. I would advise to call it "requiredConnections" or something.

Also, you might want to consider using an object when identifying functions to avoid having to use array indexes.

See (https://github.com/caolan/async/blob/v1.5.2/README.md#seriestasks-callback)